modules/loader/src/main/java/com/opengamma/strata/loader/csv/FxSingleTradeCsvLoader.java

```diff
@@ -61,43 +74,66 @@ static FxSingleTrade parse(CsvRow row, TradeInfo info, TradeCsvInfoResolver reso
    * @return the parsed trade, as an instance of {@link FxSingleTrade}
    */
   private static FxSingleTrade parseRow(CsvRow row, TradeInfo info, TradeCsvInfoResolver resolver) {
-    LocalDate paymentDate = LoaderUtils.parseDate(row.getField(PAYMENT_DATE_HEADER));
-    PayReceive leg1Direction = LoaderUtils.parsePayReceive(row.getField(LEG_1_DIRECTION_HEADER));
-    Currency leg1Currency = Currency.of(row.getField(LEG_1_CURRENCY_HEADER));
-    double leg1Notional = LoaderUtils.parseDouble(row.getField(LEG_1_NOTIONAL_HEADER));
-    PayReceive leg2Direction = LoaderUtils.parsePayReceive(row.getField(LEG_2_DIRECTION_HEADER));
-    Currency leg2Currency = Currency.of(row.getField(LEG_2_CURRENCY_HEADER));
-    double leg2Notional = LoaderUtils.parseDouble(row.getField(LEG_2_NOTIONAL_HEADER));
-    if (leg1Direction.equals(leg2Direction)) {
-      throw new IllegalArgumentException(Messages.format(
-          "Detected two legs having the same direction: {}, {}.",
-          leg1Direction.toString(),
-          leg2Direction.toString()));
+    Optional<BusinessDayAdjustment> paymentAdj = parsePaymentDateAdjustment(row);
+    if (row.findValue(BUY_SELL_FIELD).isPresent()) {
+      // convention-based
+      // ideally we'd use the trade date plus "period to start" to get the spot/payment date
+      // but we don't have all the data and it gets complicated in places like TRY, RUB and AED
+      CurrencyPair pair = CurrencyPair.parse(row.getValue(CONVENTION_FIELD));
+      BuySell buySell = LoaderUtils.parseBuySell(row.getValue(BUY_SELL_FIELD));
+      Currency currency = Currency.parse(row.getValue(CURRENCY_FIELD));
+      double notional = LoaderUtils.parseDouble(row.getValue(NOTIONAL_FIELD));
+      double fxRate = LoaderUtils.parseDouble(row.getValue(FX_RATE_FIELD));
+      LocalDate paymentDate = LoaderUtils.parseDate(row.getValue(PAYMENT_DATE_FIELD));
+      CurrencyAmount amount = CurrencyAmount.of(currency, buySell.normalize(notional));
+      FxSingle fx = paymentAdj
+          .map(adj -> FxSingle.of(amount, FxRate.of(pair, fxRate), paymentDate, adj))
+          .orElse(FxSingle.of(amount, FxRate.of(pair, fxRate), paymentDate));
```

Comment:
Probably not a huge performance gain, but using ```orElseGet(...)``` would only get the value lazily instead of computing it every time.